cpp/include/raft/spatial/knn/detail/ann_utils.cuh

```diff
@@ -151,17 +151,18 @@ inline void memzero(T* ptr, size_t n_elems, rmm::cuda_stream_view stream)
   }
 }
 
+template <typename LabelT>
 __global__ void argmin_along_rows_kernel(uint32_t n_rows,
                                          uint32_t n_cols,
                                          const float* a,
-                                         uint32_t* out)
+                                         LabelT* out)
 {
-  __shared__ uint32_t shm_ids[1024];  // NOLINT
-  __shared__ float shm_vals[1024];    // NOLINT
+  __shared__ LabelT shm_ids[1024];  // NOLINT
```

Comment:
This probably will break as soon as we more than one instantiation of the template. Consider changing the type to `uint8_t` and then `reinterpret_cast` it to `LabelT`.